Lib/fontbakery/profiles/ufo_sources.py

```diff
@@ -303,5 +307,48 @@ def com_google_fonts_check_designspace_has_consistent_codepoints(designSpace, co
         yield PASS, "Unicode assignments were consistent."
 
 
+@check(
+    id="com.thetypefounders/check/features_default_languagesystem",
+    conditions=["ufo_font"],
+    rationale="""
+        The feature file specification strongly recommends to use a
+        `languagesystem DFLT dflt` statement in your feature file. This
+        statement is automatically inserted when no `languagesystem`
+        statements are present in the feature file, *unless* there is
+        another `languagesystem` statement already present. If this is
+        the case, this behaviour could lead to unintended side effects.
+
+        This check only WARNs when this happen as there are cases where
+        not having a `languagesystem DFLT dflt` statement in your feature
+        file is technically correct.
+
+        http://adobe-type-tools.github.io/afdko/OpenTypeFeatureFileSpecification.html#4b-language-system
+    """,
+    proposal="https://github.com/googlefonts/fontbakery/issues/4011",
+)
+def com_thetypefounders_check_features_default_languagesystem(ufo_font):
+    """Check that languagesystem DFLT dflt is present in the features.fea file."""
+
+    if ufo_font.features.text is None:
+        yield SKIP, "No features.fea file in font."
+    elif not ufo_font.features.text.strip():
+        yield PASS, "Default languagesystem inserted by compiler."
+    else:
+        tags = re.findall(
+            # pylint: disable-next=line-too-long
```

Comment:
I tried a couple of different variations to keep the line shorter, but they all became harder to read/understand. Happy to change if anyone has suggestions...